redisinsight/ui/src/services/workbenchStorage.ts

```diff
@@ -0,0 +1,350 @@
+        })
+      } catch (e) {
+        reject(e)
+      }
+    })
+  }
+
+  /**
+   * Parses the given key by normalizing it and splitting it using the key separator.
+   * Key can consist of <storeName>:<keyId> potentially more items
+   * @param {unknown} key - The key to be parsed.
+   */
+  private _parseKey(key: unknown) {
+    const nKey = this._normalizeKey(key)
+
+    const [storeName, keyName] = nKey.split(this.keySeparator)
+    return {
+      storeName,
+      keyName,
+    }
+  }
+
+  private _normalizeKey(key: unknown) {
+    // Cast the key to a string, as that's all we can set as a key.
+    if (typeof key !== 'string') {
+      console.warn(`${key} used as a key, but it is not a string.`)
+      return String(key)
+    }
+
+    return key
+  }
+}
+
+export const wbHistoryStorage = new WorkbenchStorage('RI_WB_HISTORY', 1)
+
+type CommandHistoryItem = {
+  dbId: string, // Instance Id
+  commandId: string,
+  data: CommandExecution[]
+}
+
+type CommandHistoryType = CommandHistoryItem[]
+
+export async function getLocalWbHistory(instanceId: string) {
+  try {
+    const key = `${BrowserStorageItem.wbCommandsHistory}:${instanceId}`
+    const history = await wbHistoryStorage.getItem(key) as CommandHistoryType
+
+    return history || []
+  } catch (e) {
+    console.error(e)
+    return []
+  }
+}
+
+export function saveLocalWbHistory(commandsHistory: CommandHistoryType) {
+  try {
+    const key = BrowserStorageItem.wbCommandsHistory
+    return Promise.all(commandsHistory.map((chItem) => wbHistoryStorage.setItem(key, chItem)))
+  } catch (e) {
+    console.error(e)
+    return null
+  }
+}
+
+async function cleanupDatabaseHistory(dbId: string) {
+  const commandsHistory: CommandHistoryType = await getLocalWbHistory(dbId)
+  let size = 0
+  const update = commandsHistory.reduce((acc, commandsHistoryElement) => {
+    if (size > riConfig.workbench.maxItemsPerDb) {
+      return acc
+    }
+    let items = commandsHistoryElement.data
+    size += items.length
+    if (size > riConfig.workbench.maxItemsPerDb) {
+      const diff = riConfig.workbench.maxItemsPerDb - size
+      items = items.slice(diff)
+      acc.push({
+        dbId,
+        data: items,
+        commandId: commandsHistoryElement.commandId
+      })
+    }
+
+    return acc
+  }, [] as CommandHistoryType)
+
+  await saveLocalWbHistory(update)
+}
+
+export async function addCommands(dbId: string, commandId: string, data: CommandExecution[]) {
+  // Store command results in local storage!
+  const commandsHistory: CommandHistoryType = await getLocalWbHistory(dbId)
+  const storedData = data.map((item) => {
+    // Do not store command execution result that exceeded limitation
+    if (JSON.stringify(item.result).length > riConfig.workbench.maxResultSize) {
+      item.result = [
+        {
+          status: CommandExecutionStatus.Success,
+          response: `Results have been deleted since they exceed ${formatBytes(riConfig.workbench.maxResultSize)}. 
```

Comment:
Looks like there's an error with this when the result is a table:
<img width="1662" alt="image" src="https://github.com/user-attachments/assets/391e881a-f545-422f-a0ad-45a87015144e">

It's still trying to render the table instead of showing this error message

Reply:
that appears to be bug in redisearch plugin